### ISSUE TITLE
Migrate all UI elements to use Nui.nvim

### ISSUE THREAD

**Issue by akinsho** [akinsho/flutter-tools.nvim#219]
Using a UI framework like [nui.nvim](https://github.com/MunifTanjim/nui.nvim) will reduce the maintenance burden of this plugin since as well as managing the LSP and finding executables we wouldn't also have to maintain a mini UI library as well and could just depend on another tool for that. nui has become quite mature and is depended on by a lot of other plugins so would hopefully end up being something a user has installed anyway, as well as all that it would allow me to simplify the functionality and responsibilities of this plugin greatly

**Comment by akinsho** [comment]
Actually following #221 it turns out that a lot of UI could just be delegated to `vim.ui.*` and so I'm happy to continue using that as the main refactoring path but for complex UI like the tree view or making splits it might be nice to use nui's UI instead.

TLDR: the focus is now on exploring using nui for complex ui mainly e.g. the outline view
